src/components/simulation/tray-image.tsx

```diff
@@ -11,38 +11,52 @@ import "./tray-image.scss";
 // corner. This ends up being about 5 pixels for all tray objects. Technically it could be computed
 // by hand for ALL tray objects, but subtracting 5 pixels works pretty well in practice.
 const kOutlineOffset = 5;
+const kNonTrayPreviewHeight = 64;
 
 interface IProps {
   trayObject: TrayObject;
   onTrayObjectSelect: (type: TrayType) => void;
   traySelectionType?: TrayType;
+  dragOverTray: boolean;
 }
 
 export const TrayImage: React.FC<IProps> = (props) => {
-  const { trayObject, onTrayObjectSelect, traySelectionType } = props;
+  const { trayObject, onTrayObjectSelect, traySelectionType, dragOverTray } = props;
   const TrayObjectImage = trayObject.image;
 
-  const [{isDragging, dragSourcePosition}, drag ] = useDrag({
+  const [{isDragging, dragSourcePosition, dragPosition}, drag ] = useDrag({
     item: { type: trayObject.type, trayIndex: trayObject.trayIndex, dragImage: trayObject.dragImage,
             left: trayObject.left, top: trayObject.top },
     collect: monitor => ({
       isDragging: !!monitor.isDragging(),
-      dragSourcePosition: monitor.getSourceClientOffset()
+      dragPosition: monitor.getClientOffset(),
+      dragSourcePosition: monitor.getSourceClientOffset(),
     }),
   });
 
   const PreviewImage = () => {
     const {display, item} = usePreview();
-    if (!display || !dragSourcePosition) {
+    if (!display || !dragSourcePosition || !dragPosition) {
       return null;
     }
     const boundingBoxDeltaX = (trayObject.boundingBoxWidth - trayObject.width) / 2;
     const boundingBoxDeltaY = (trayObject.boundingBoxHeight - trayObject.height) / 2;
-    const previewStyle = {
-      left: dragSourcePosition.x + boundingBoxDeltaX - kOutlineOffset,
-      top: dragSourcePosition.y + boundingBoxDeltaY - kOutlineOffset,
-      transform: `rotate(${trayObject.rotation}deg)`
-    };
+    let previewStyle = {};
+    const isLeaf = draggableLeafTypes.includes(trayObject.type as any);
+    if (dragOverTray || isLeaf) {
+      previewStyle = {
+        left: dragSourcePosition.x + boundingBoxDeltaX - kOutlineOffset,
+        top: dragSourcePosition.y + boundingBoxDeltaY - kOutlineOffset,
+        transform: `rotate(${trayObject.rotation}deg)`
+      };
+    } else {
+      const previewHeight = Math.min(kNonTrayPreviewHeight, trayObject.height);
+      previewStyle = {
+        height: previewHeight,
+        left: dragPosition.x - previewHeight / trayObject.height * trayObject.width / 2,
```

Comment:
```suggestion
        left: dragPosition.x - (previewHeight / trayObject.height) * trayObject.width / 2,
```
parentheses aren't required, but order-of-operation takes a bit of study to work out without them. It might even make sense to define a local `scale` value to make the intent clear.

Reply:
Yeah, I like adding these sorts of additional parens if it makes things easier to read.  Fixed.